File: chroma-league/src/main/java/net/booone/chromaleague/tasks/CheckRiotApiTask.java
package net.booone.chromaleague.tasks;

import net.booone.chromaleague.state.RunningState;

import java.net.InetSocketAddress;
import java.net.Socket;

public class CheckRiotApiTask implements Runnable {
    private static final int RIOT_API_PORT = 2999;

    @Override
    public void run() {
        RunningState.setRiotApi(checkRiotApiUp());
    }

    public static Boolean checkRiotApiUp() {
        try (Socket socket = new Socket()) {
            socket.connect(new InetSocketAddress("localhost", RIOT_API_PORT), 100);
            return Boolean.TRUE;
        } catch (Exception e) {
            return Boolean.FALSE;
        }
    }
}

File: src/tasks/CheckRiotApiTask.ts
import {RunningState} from '../state/RunningState.js';
import {checkRiotApiUp} from './RiotApiChecker.js';

const FAILURES_BEFORE_DISCONNECT = 3;

export class CheckRiotApiTask {
    private consecutiveFailures = 0;

    async run(): Promise<void> {
        const isUp = await checkRiotApiUp();
        if (isUp) {
            this.consecutiveFailures = 0;
            RunningState.setRiotApi(true);
        } else {
            this.consecutiveFailures++;
            if (this.consecutiveFailures >= FAILURES_BEFORE_DISCONNECT) {
                RunningState.setRiotApi(false);
            }
        }
    }
}
